### PR TITLE
Don't embed c runtime library in static lib

### PR DESCRIPTION
**Pull request by saolsen**
Currently the musl runtime is embedded in our static lib for linux. This works if the kernel it's run on is compatible with the musl version but doesn't work otherwise.
Now we allow the application to link dynamically with the c runtime library and that should work on more platforms.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
       - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
     branches:
       - main
+      - steve/gooooo
```
Comment (samscott89) [inline .github/workflows/release.yml:8]:
```suggestion
```
Reminder to delete before merge

**Review by samscott89 — APPROVED** [review]

File: docs/content/any/project/changelogs/NEXT.md [hunk 2]
```diff
@@ -11,6 +11,15 @@ draft: true
 
 ## `RELEASED_PACKAGE_1` NEW_VERSION
 
+### Core
+
+#### Other bugs & improvements
+
+- Fixed the way we build our static library on Linux so it doesn't embed
+  musl and instead links to the system c runtime library.
+  Language that depend on the static lib Linux build such as python and go
```
Comment (gj) [inline docs/content/any/project/changelogs/NEXT.md:20]:
```suggestion
  Languages that depend on the static lib Linux build such as python and go
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -497,6 +497,10 @@ jobs:
         if: runner.os == 'Linux' && matrix.go-version != '1.12'
         run: docker run --rm -v `pwd`:/oso tetafro/golang-gcc:${{ matrix.go-version }}-alpine /bin/sh -c 'cd /oso && go build && go test -v ./tests/'
         working-directory: go
+      - name: "test buster"
+        if: runner.os == 'Linux' && matrix.go-version != '1.12'
```
Comment (gj) [inline .github/workflows/release.yml:501]: Do we need to skip 1.12 here?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -497,6 +497,10 @@ jobs:
...
+        run: docker run --rm -v `pwd`:/oso golang:${{ matrix.go-version }}-buster /bin/sh -c 'cd /oso && go build && go test -v ./tests/'
```
Comment (gj) [inline .github/workflows/release.yml:502]: Is `go build` necessary before `go test`?

**Pull request merged**